ucsmsdk/ucscoreutils.py

```diff
@@ -496,17 +501,28 @@ def __init__(self, class_id):
         self.children = self.__mo_meta.children
         self.props = {}
 
-    def get_prop_dict(self):
-        class_obj = load_class(self.class_id)
-        self.props = class_obj.prop_meta
+        self._str_tree = "\n"
+        self._str_props = "\n"
+
+        if show_tree:
+            self._str_tree = _show_tree(class_id, depth)
+
+        if include_prop:
+            class_obj = load_class(self.class_id)
+            self.props = class_obj.prop_meta
+            for prop in sorted(self.props):
+                self._str_props += str(self.props[prop]) + "\n"
 
     def __str__(self):
         """
         Method to return string representation.
         """
-
         ts = 8
-        out_str = "\n"
+        out_str = ""
+
+        out_str += self._str_tree
```

Comment:
as I said earlier, no need to check when the default value is "\n". It the option was false, it will print a newline and continue, otherwise it will have the complete string that needs to be printed.